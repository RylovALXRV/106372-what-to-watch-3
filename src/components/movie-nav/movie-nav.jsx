import React, {Fragment, PureComponent} from "react";
import PropTypes from "prop-types";
import Overview from "../overview/overview.jsx";
import Details from "../details/details.jsx";
import Reviews from "../reviews/reviews.jsx";

const MovieNavLink = [Overview, Details, Reviews];

const MOVIE_NAV_LINKS = [
  `Overview`,
  `Details`,
  `Reviews`,
];

class MovieNav extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activeNavLinkId: 0,
    };
  }

  _renderMovieDescription() {
    const Component = MovieNavLink[this.state.activeNavLinkId];

    return <Component movie={this.props} />;
  }

  render() {
    return (
      <Fragment>
        <nav className="movie-nav movie-card__nav">
          <ul className="movie-nav__list">
            {MOVIE_NAV_LINKS.map((movieNavLink, i) => (
              <li
                className={`movie-nav__item ${i === this.state.activeNavLinkId ? `movie-nav__item--active` : ``}`}
                key={movieNavLink}
              >
                <a
                  href="#"
                  className="movie-nav__link"
                  onClick={(evt) => {
                    evt.preventDefault();
                    this.setState({
                      activeNavLinkId: i,
                    });
                  }}
                >{movieNavLink}</a>
              </li>
            ))}
          </ul>
        </nav>
        {this._renderMovieDescription()}
      </Fragment>
    );
  }
}

MovieNav.propTypes = {
  genre: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  director: PropTypes.string.isRequired,
  duration: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(
      PropTypes.string.isRequired
  ).isRequired,
  reviews: PropTypes.arrayOf(
      PropTypes.exact({
        text: PropTypes.string.isRequired,
        rating: PropTypes.number.isRequired,
        author: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
      }).isRequired
  ).isRequired,
  descriptions: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
};

export default MovieNav;
