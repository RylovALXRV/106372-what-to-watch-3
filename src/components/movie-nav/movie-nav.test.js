import React from "react";
import renderer from "react-test-renderer";
import {MovieNav} from "./movie-nav.jsx";

const mock = {
  director: `David Yates`,
  starring: [`Rami Malek`, `Jon Finch`, `Cate Blanchett`],
  duration: `1h 30m`,
  genre: `drama`,
  year: 2018,
  reviews: [{
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: 8.0,
    author: `Kate Muir`,
    date: `December 20, 2016`,
  }],
  descriptions: `In late 1823, Hugh Glass guides Captain Andrew Henry's trappers through territory of the present day Dakotas.`,
  rating: 8.0,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
};

describe(`Render MovieNav`, () => {
  it(`Render MovieNav with Overview`, () => {
    const tree = renderer.create(
        <MovieNav
          card={mock}
          onMovieNavClick={() => {}}
          movieNavLinkIndex={0}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieNav with Details`, () => {
    const tree = renderer.create(
        <MovieNav
          card={mock}
          onMovieNavClick={() => {}}
          movieNavLinkIndex={1}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieNav with Reviews`, () => {
    const tree = renderer.create(
        <MovieNav
          card={mock}
          onMovieNavClick={() => {}}
          movieNavLinkIndex={2}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
