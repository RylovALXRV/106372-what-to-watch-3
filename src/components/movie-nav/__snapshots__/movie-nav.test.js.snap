// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieNav Render MovieNav with Details 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          David Yates
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Rami Malek, 
Jon Finch, 
Cate Blanchett
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          1h 30m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          drama
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2018
        </span>
      </p>
    </div>
  </div>,
]
`;

exports[`Render MovieNav Render MovieNav with Overview 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        240 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      In late 1823, Hugh Glass guides Captain Andrew Henry's trappers through territory of the present day Dakotas.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        David Yates
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Rami Malek, Jon Finch, Cate Blanchett
         and other
      </strong>
    </p>
  </div>,
]
`;

exports[`Render MovieNav Render MovieNav with Reviews 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Kate Muir
            </cite>
            <time
              className="review__date"
              dateTime="December 20, 2016"
            >
              December 20, 2016
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    />
  </div>,
]
`;
